# Importimised

In [1]:
import pandas as pd
import matplotlib.pyplot as plt


In [15]:
#!pip install pandas openpyxl

# sisselugemised

In [2]:
#excel file 1
EEB_samples = pd.ExcelFile('EEB_samples.xlsx') #file

sample = EEB_samples.parse('full_table') #sheet name
#samples_reynoutria = EEB_samples.parse('reynoutria') 
#samples_solidago = EEB_samples.parse('solidago') 

#excel file 2
EEB_soil_chemistry = pd.ExcelFile('EEB_soil_chemistry.xlsx')

soil = EEB_soil_chemistry.parse('Sheet1')

#excel file 3
plant_survey = pd.ExcelFile('Plant_survey_2023_final.xlsx') 

plant = plant_survey.parse('Sheet1')

### eemaldame soil failist ühikute rea

In [3]:
soil=soil.iloc[1:] #eemaldasin ühikute rea
soil= soil.reset_index(drop=True)

# asjalik

In [4]:
sample.head()

new_code project plant_sp sampling_area(on_bag) treatment  \
0  EEB-RJ1-1     EEB       RJ               Mustvee   control   
1  EEB-RJ1-2     EEB       RJ               Mustvee   control   
2  EEB-RJ1-3     EEB       RJ               Mustvee   control   
3  EEB-RJ1-4     EEB       RJ               Mustvee   control   
4  EEB-RJ1-5     EEB       RJ               Mustvee   control   

   sampling_area_nr      gradient  grad_nr  plant_nr expert area        lat  \
0                 1       invaded        1         1  Kadri    E  58.858114   
1                 1       invaded        2         2  Kadri    E  58.858114   
2                 1       invaded        3         3  Kadri    E  58.858114   
3                 1  transitional        1         4  Kadri    E  58.858114   
4                 1  transitional        2         5  Kadri    E  58.858114   

         lon status _dna_extraction  
0  26.931854                    NaN  
1  26.931854                    NaN  
2  26.931854                    NaN  
3  26.931854                    NaN  
4  26.931854                    NaN

In [5]:
soil.head()

ala_kood         Proovi nimetus  Nr.  Jrk. nr.  pHKCl        P-AL  \
0  EEB-SC2          Raadi_sol_can  1.0      21.0   6.98  110.955897   
1  EEB-SC6   Rõngu_sol_can_mowing  2.0      22.0   5.98   58.659218   
2  EEB-SC5    Rõngu_sol_can_steam  3.0      23.0   6.91   78.558771   
3  EEB-RJ4  Sangaste_Rey_Jap_cont  4.0      24.0   6.68   122.55489   
4  EEB-RJ1   Mustvee_Rey_Jap_cont  5.0      25.0   6.30    3.324208   

         K-AL        Ca-AL  Mg-AL N -Kjeldahl  C-Tjurin  
0  122.131311  9938.136101  491.4    0.277187  3.061475  
1  163.407821  1006.783719  108.8    0.131683  1.432409  
2  113.998819  3593.227013  563.8    0.129682  1.280281  
3   84.231537  4658.483034  296.3    0.330927  3.975705  
4  159.170903  2712.749316  143.1    0.259012  3.017905

In [6]:
plant.head()

sample  Reynoutria_japonica  Solidago_canadensis  Aegopodium_podagraria  \
0  EEB-RJ5-1                   60                    0                    5.0   
1  EEB-RJ5-2                   60                    0                    1.0   
2  EEB-RJ5-3                   90                    0                    5.0   
3  EEB-RJ5-4                   20                    0                   80.0   
4  EEB-RJ5-5                   10                    0                   70.0   

   Ranunculus_auricomus  Alopecurus_pratensis  Elymus_repens  \
0                   0.5                   0.0            0.0   
1                   0.5                   0.0            0.0   
2                   2.0                   0.0            0.0   
3                   1.0                   1.0            3.0   
4                   0.0                   0.0            3.0   

   Agrostis_capillaris  Festuca_rubra  Cirsium_arvensis  ...  \
0                  0.0            0.0               0.0  ...   
1                  0.0            0.0               0.0  ...   
2                  0.0            0.0               0.0  ...   
3                  0.5            0.0               0.0  ...   
4                  0.5            0.5               0.5  ...   

   Plantago_lanceolata  Daucus_carota  Achillea_millefolia  \
0                    0              0                    0   
1                    0              0                    0   
2                    0              0                    0   
3                    0              0                    0   
4                    0              0                    0   

   Allium_scorodoprasum  Veronica_teucrium  Agrimonia_eupatoria  \
0                     0                  0                    0   
1                     0                  0                    0   
2                     0                  0                    0   
3                     0                  0                    0   
4                     0                  0                    0   

   Plantago_media  Sonchus_arvensis  Prunus_domestica  Salix_alba  
0               0                 0                 0           0  
1               0                 0                 0           0  
2               0                 0                 0           0  
3               0                 0                 0           0  
4               0                 0                 0           0  

[5 rows x 222 columns]

In [7]:
soil.shape

(24, 11)

In [8]:
sample.shape

(216, 14)

In [9]:
plant.shape

(216, 222)

# keskmised

 - 123 inv
 - 456trans
 - 789nat

EEB-RJ1-1  

RJ taga indeks on 1 kuni 13


EEB-SC1-1

SC taga indeks 1 kuni 11


viimane indeks 1-9 kõigil

In [10]:
plant[plant['sample']=='EEB-RJ5-1']

sample  Reynoutria_japonica  Solidago_canadensis  Aegopodium_podagraria  \
0  EEB-RJ5-1                   60                    0                    5.0   

   Ranunculus_auricomus  Alopecurus_pratensis  Elymus_repens  \
0                   0.5                   0.0            0.0   

   Agrostis_capillaris  Festuca_rubra  Cirsium_arvensis  ...  \
0                  0.0            0.0               0.0  ...   

   Plantago_lanceolata  Daucus_carota  Achillea_millefolia  \
0                    0              0                    0   

   Allium_scorodoprasum  Veronica_teucrium  Agrimonia_eupatoria  \
0                     0                  0                    0   

   Plantago_media  Sonchus_arvensis  Prunus_domestica  Salix_alba  
0               0                 0                 0           0  

[1 rows x 222 columns]

In [11]:
for i in range (1,14):
    for j in range(1,10):
        plant[plant['sample']=='EEB-RJ{}-{}'.format(i, j)]
        if j in range(1, 4):
            plant.loc[plant['sample'] == 'EEB-RJ{}-{}'.format(i, j), 'sample'] = 'R-INV{}'.format(i)
        if j in range(4, 7): 
            plant.loc[plant['sample'] == 'EEB-RJ{}-{}'.format(i, j), 'sample'] = 'R-TRANS{}'.format(i)
        if j in range(7, 10): 
            plant.loc[plant['sample'] == 'EEB-RJ{}-{}'.format(i, j), 'sample'] = 'R-NAT{}'.format(i)
            
            
for i in range (1,12):
    for j in range(1,10):
        plant[plant['sample']=='EEB-SC{}-{}'.format(i, j)]
        if j in range(1, 4):
            plant.loc[plant['sample'] == 'EEB-SC{}-{}'.format(i, j), 'sample'] = 'S-INV{}'.format(i)
        if j in range(4, 7): 
            plant.loc[plant['sample'] == 'EEB-SC{}-{}'.format(i, j), 'sample'] = 'S-TRANS{}'.format(i)
        if j in range(7, 10): 
            plant.loc[plant['sample'] == 'EEB-SC{}-{}'.format(i, j), 'sample'] = 'S-NAT{}'.format(i)

In [12]:
plant.tail(20)

sample  Reynoutria_japonica  Solidago_canadensis  \
196     S-NAT9                    0                    0   
197     S-NAT9                    0                    0   
198    R-INV13                  100                    0   
199    R-INV13                   75                    0   
200    R-INV13                   50                    0   
201  R-TRANS13                    3                    0   
202  R-TRANS13                    5                    0   
203  R-TRANS13                   10                    0   
204    R-NAT13                    0                    0   
205    R-NAT13                    0                    0   
206    R-NAT13                    0                    0   
207    R-INV11                   85                    0   
208    R-INV11                   90                    0   
209    R-INV11                  100                    0   
210  R-TRANS11                    7                    0   
211  R-TRANS11                    7                    0   
212  R-TRANS11                   12                    0   
213    R-NAT11                    0                    0   
214    R-NAT11                    0                    0   
215    R-NAT11                    0                    0   

     Aegopodium_podagraria  Ranunculus_auricomus  Alopecurus_pratensis  \
196                   20.0                   0.0                   1.0   
197                    0.0                   0.0                   0.0   
198                    4.0                   0.0                   0.0   
199                    0.0                   0.0                   0.0   
200                    0.0                   0.0                   0.0   
201                   99.0                   0.0                   0.0   
202                  100.0                   0.0                   0.0   
203                   99.0                   0.0                   0.0   
204                  100.0                   0.0                   0.0   
205                  100.0                   0.0                   0.0   
206                   90.0                   0.0                   0.0   
207                    0.0                   0.0                   0.0   
208                    8.0                   0.0                   0.0   
209                    0.0                   0.0                   0.0   
210                   75.0                   0.0                   0.0   
211                  100.0                   0.0                   0.0   
212                    6.0                   0.0                   0.0   
213                   98.0                   0.0                   0.0   
214                  100.0                   0.0                   0.0   
215                    0.0                   0.0                   0.0   

     Elymus_repens  Agrostis_capillaris  Festuca_rubra  Cirsium_arvensis  ...  \
196            0.0                  0.0            9.0               0.0  ...   
197            0.0                  0.0           35.0               0.0  ...   
198            0.0                  0.0            0.0               0.0  ...   
199            0.0                  0.0            0.0               0.0  ...   
200            0.0                  0.0            0.0               0.0  ...   
201            1.0                  0.0            0.0               0.0  ...   
202            1.0                  0.0            0.0               0.0  ...   
203            1.0                  0.0            0.0               0.0  ...   
204           10.0                  0.0            0.0               0.0  ...   
205            2.0                  0.0            0.0               0.0  ...   
206            1.0                  0.0            0.0               0.0  ...   
207            0.0                  0.0            0.0               0.0  ...   
208            0.0                  0.0            0.0               0.0  ...   
209            0.0                  0.0            0.0    

In [18]:
mean_df = plant.groupby('sample').mean()

In [19]:
mean_df

Reynoutria_japonica  Solidago_canadensis  Aegopodium_podagraria  \
sample                                                                      
R-INV1              86.666667             0.000000              18.333333   
R-INV10             71.666667             0.000000               0.000000   
R-INV11             91.666667             0.000000               2.666667   
R-INV12             90.666667             0.000000              99.666667   
R-INV13             75.000000             0.000000               1.333333   
...                       ...                  ...                    ...   
S-TRANS5             0.000000            26.666667               0.000000   
S-TRANS6             0.000000             6.666667               0.000000   
S-TRANS7             0.000000            25.000000               0.000000   
S-TRANS8             0.000000            30.000000              30.000000   
S-TRANS9             0.000000            11.333333               0.666667   

          Ranunculus_auricomus  Alopecurus_pratensis  Elymus_repens  \
sample                                                                
R-INV1                0.000000              0.000000       0.000000   
R-INV10               0.000000              0.000000       0.000000   
R-INV11               0.000000              0.000000       0.000000   
R-INV12               0.000000              0.000000       0.000000   
R-INV13               0.000000              0.000000       0.000000   
...                        ...                   ...            ...   
S-TRANS5              0.000000              2.000000       0.000000   
S-TRANS6              0.000000              8.333333       0.000000   
S-TRANS7              0.000000              0.000000       4.333333   
S-TRANS8              1.666667              0.333333       1.666667   
S-TRANS9              0.000000              0.666667       0.000000   

          Agrostis_capillaris  Festuca_rubra  Cirsium_arvensis  \
sample                                                           
R-INV1               0.000000       0.000000               0.0   
R-INV10              0.000000       0.000000               0.0   
R-INV11              0.000000       0.000000               0.0   
R-INV12              0.000000       0.000000               0.0   
R-INV13              0.000000       0.000000               0.0   
...                       ...            ...               ...   
S-TRANS5             0.000000      18.333333               0.0   
S-TRANS6             0.000000       0.000000               0.0   
S-TRANS7             2.333333       5.000000               0.0   
S-TRANS8             0.000000       0.000000               0.0   
S-TRANS9             0.000000       1.666667               0.0   

          Taraxacum_officinale  ...  Plantago_lanceolata  Daucus_carota  \
sample                          ...                                       
R-INV1                0.000000  ...                  0.0       0.000000   
R-INV10               0.000000  ...                  0.0       0.000000   
R-INV11               0.333333  ...                  0.0       0.000000   
R-INV12               0.000000  ...                  0.0       0.000000   
R-INV13               0.666667  ...                  0.0       0.000000   
...                        ...  ...                  ...            ...   
S-TRANS5              0.000000  ...                  0.0       0.000000   
S-TRANS6              0.333333  ...                  0.0       0.000000   
S-TRANS7              0.000000  ...                  0.0       0.000000   
S-TRANS8              0.000000  ...                  0.0       0.000000   
S-TRANS9              1.000000  ...                  0.0       0.333333   

          Achillea_millefolia  Allium_scorodoprasum  Veronica_teucrium  \
sample                                                                   
R-INV1                    0.0              0.000000           0.000000   
R-INV10                   0